x-pack/plugins/spaces/server/spaces_service/spaces_service.ts

```diff
@@ -4,126 +4,123 @@
 import { DEFAULT_SPACE_ID } from '../../common/constants';
 import { spaceIdToNamespace, namespaceToSpaceId } from '../lib/utils/namespace';
-import { Space } from '../../common/model/space';
-import { SpacesAuditLogger } from '../lib/audit_logger';
-
-type RequestFacade = KibanaRequest | Legacy.Request;
+import { Space } from '..';
 
 export interface SpacesServiceSetup {
-  scopedClient(request: RequestFacade): Promise<SpacesClient>;
-
-  getSpaceId(request: RequestFacade): string;
-
-  getBasePath(spaceId: string): string;
+  /**
+   * Retrieves the space id associated with the provided request.
+   * @param request
+   *
+   * @deprecated Use `getSpaceId` from the `SpacesServiceStart` contract instead.
+   */
+  getSpaceId(request: KibanaRequest): string;
+
+  /**
+   * Converts the provided space id into the corresponding Saved Objects `namespace` id.
+   * @param spaceId
+   *
+   * @deprecated use `spaceIdToNamespace` from the `SpacesServiceStart` contract instead.
+   */
+  spaceIdToNamespace(spaceId: string): string | undefined;
 
-  isInDefaultSpace(request: RequestFacade): boolean;
+  /**
+   * Converts the provided namespace into the corresponding space id.
+   * @param namespace
+   *
+   * @deprecated use `namespaceToSpaceId` from the `SpacesServiceStart` contract instead.
+   */
+  namespaceToSpaceId(namespace: string | undefined): string;
+}
 
+export interface SpacesServiceStart {
+  /**
+   * Creates a scoped instance of the SpacesClient.
+   */
+  createSpacesClient: SpacesClientServiceStart['createSpacesClient'];
+
+  /**
+   * Retrieves the space id associated with the provided request.
+   * @param request
+   */
+  getSpaceId(request: KibanaRequest): string;
+
+  /**
+   * Indicates if the provided request is executing within the context of the `default` space.
+   * @param request
+   */
+  isInDefaultSpace(request: KibanaRequest): boolean;
+
+  /**
+   * Retrieves the Space associated with the provided request.
+   * @param request
+   */
+  getActiveSpace(request: KibanaRequest): Promise<Space>;
+
+  /**
+   * Converts the provided space id into the corresponding Saved Objects `namespace` id.
+   * @param spaceId
+   */
   spaceIdToNamespace(spaceId: string): string | undefined;
 
+  /**
+   * Converts the provided namespace into the corresponding space id.
+   * @param namespace
+   */
   namespaceToSpaceId(namespace: string | undefined): string;
+}
 
-  getActiveSpace(request: RequestFacade): Promise<Space>;
+interface SpacesServiceSetupDeps {
+  basePath: IBasePath;
 }
 
-interface SpacesServiceDeps {
-  http: CoreSetup['http'];
-  getStartServices: CoreSetup['getStartServices'];
-  authorization: SecurityPluginSetup['authz'] | null;
-  config$: Observable<ConfigType>;
-  auditLogger: SpacesAuditLogger;
+interface SpacesServiceStartDeps {
+  basePath: IBasePath;
+  spacesClientService: SpacesClientServiceStart;
 }
 
 export class SpacesService {
   private configSubscription$?: Subscription;
```

Comment:
__nit:__ looks like we don't need this in this service anymore, and we may want to move this to `SpacesClientService` instead?